### PR TITLE
Turn logic

### PR DESCRIPTION
**Pull request by PaleoGuy2012**
Added some turn logic and also the move_to stuff that I mentioned last night.  I deleted the other PR since this has all of that in it.  Still can't get the pieces to go to the right spots.  I have the is_obstructed logic still commented out in all of the piece models until we get that totally squared away.  --Ryan

### CODE REVIEW

File: app/controllers/pieces_controller.rb [hunk 1]
```diff
@@ -10,9 +10,9 @@ def create
 
   def update
     @piece = Piece.find(params[:id])
-    @piece.move_to!(:x_position, :y_position) 
-    @game.update(white_player_turn: !@game.white_player_turn)
-    redirect_to game_path(@game)
+    @piece.move_to!(params[:x_position], params[:y_position])
+    redirect_to game_path(current_game[:id])
+    @current_game.update(white_player_turn: !@current_game.white_player_turn)
```
Comment (ilyakrasnov) [inline app/controllers/pieces_controller.rb]: Nice job @PaleoGuy2012 

I'd suggest to move the whole logic into the `move_to` method and add a method on the game model.. sth like `switch_turns!` which do whatever the line 15 does.. and in the `move_to` of the piece you can just say: `game.switch_turns!` hope it makes sense

File: app/models/knight.rb [hunk 2]
```diff
@@ -2,14 +2,16 @@ class Knight < Piece
...
+	  return false if square_occupied?(new_y, new_y)
 	  return false if off_board?(new_x, new_y)
 	  return true if 
 	  	(x_position - new_x.to_i).abs == 1 && (y_position - new_y.to_i).abs == 2 ||
 	  	(x_position - new_x.to_i).abs == 2 && (y_position - new_y.to_i).abs == 1  
 	end 
 	  false 
 end
+
+KNIGHT = 'Knight'.freeze
```
Comment (ilyakrasnov) [inline app/models/knight.rb]: Not sure what the purpose of the frozen strings is...

File: app/views/pieces/show.html.erb [hunk 3]
```diff
@@ -1,19 +1,22 @@
-
-
-
-
-
 <h1 class="text-center">The Pawnshop</h1>
 
-  
+  <div id="turn">
+      <% if current_game.white_player_turn == true %>
```
Comment (ilyakrasnov) [inline app/views/pieces/show.html.erb]: we could refactor it to one line saying: 

`<p>The current turn belongs to: current_game.player_at_turn %></p>`

And on the game model you can define the method `player_at_turn` (maybe you can find a better name 😄 ) and move the logic over there...

File: app/controllers/pieces_controller.rb (hunk 1 above)
Comment (PaleoGuy2012) [reply app/controllers/pieces_controller.rb]: That does, I'll work on that good suggestion.  Keep those controllers skinny.

File: app/models/knight.rb (hunk 2 above)
Comment (PaleoGuy2012) [reply app/models/knight.rb]: Left you a message about this in the slack, but I found this freeze method to be able to reference constants elsewhere in the app, such as the check logic with the King.  What's a better way that I can go about this?

File: app/views/pieces/show.html.erb (hunk 3 above)
Comment (PaleoGuy2012) [reply app/views/pieces/show.html.erb]: Ok I'll look at cleaning this up thank you.

**Pull request merged**